### DISCUSSION TITLE
Tooltip and DialogTrigger

### DISCUSSION THREAD

**Discussion by martpet** [adobe/react-spectrum#2714] [Q&A]
Hello,
what would be the correct way to add a tooltip to an icon that opens a dialog?

I'm trying with this code:

```
<TooltipTrigger>
  <DialogTrigger type="popover" isOpen={open} onOpenChange={setOpen}>
    <ActionButton isQuiet aria-label={formatMessage({ id: 'button.settings' })}>
      <SettingsIcon />
    </ActionButton>
    <Dialog size="L">
      <Content>
        <Settings />
      </Content>
    </Dialog>
  </DialogTrigger>
  <Tooltip>My tooltip text</Tooltip>
</TooltipTrigger>
```

Hovering over the icon when the dialog is closed is OK:

![Screenshot 2021-12-26 at 21 40 28](https://user-images.githubusercontent.com/57956816/147418632-8f3f7342-04db-4b68-8d5c-dc824d21c1c4.png)

However when the dialog is open and hovering over the icon, the tooltip appears over a Picker inside the dialog:

![Screenshot 2021-12-26 at 21 52 22](https://user-images.githubusercontent.com/57956816/147418684-78037e6b-d783-4f4b-9f1d-36a0214dc479.png)

**Comment by snowystinger** [comment]
Ah, Tooltip context would be going to the Picker probably if you wrap the whole Dialog in a tooltip trigger.

Better instead to only wrap the ActionButton with the TooltipTrigger and then put the DialogTrigger around all of that.

**Comment by martpet** [comment]
Hey, snowystinger, thank you for your answer. I tried it but the dialog won't open if I do it like this:

```
<DialogTrigger type="popover" isOpen={open} onOpenChange={setOpen}>
  <TooltipTrigger>
    <ActionButton>
      <SettingsIcon />
    </ActionButton>
    <Tooltip>My tooltip</Tooltip>
  </TooltipTrigger>
  <Dialog>
    <Content>
    </Content>
  </Dialog>
</DialogTrigger>
```

**Reply by snowystinger** [reply]
Ah yep, I see that here https://codesandbox.io/s/youthful-lamarr-up3ur?file=/src/App.js
Possibly a bug, could you open an issue? We are on shutdown right now, so won't get to really look at it until the new year. You're welcome to make a PR as well to address it. I'd probably compare ActionMenu and DialogTriggers. Though our Menu's may have the same issue, but since they don't have a Picker inside them, we don't see this issue.
